ls1intum/Artemis
# `Programming exercises`: Fix Jenkins builds by replacing all non-printable characters in JUnit-XMLs

### Checklist
#### General


#### Changes affecting Programming Exercises
~~- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).~~

### Motivation and Context
Closes #6149

### Description
In #5780 the same fix has been applied to only C exercises. This fix is now applied to all exercise types.

### Steps for Testing
Prerequisites:
- TS2 or local setup with Jenkins
- 1 Programming exercise

1. Add a test-case to the Tests-Repository of the exercise so that the failure message contains a non-printable character.
Example for Java:
```java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class DummyTest {
    @Test
    void nonPrintable() {
        assertEquals("printable", "non-print\u007Fable");
    }
}
```
2. Artemis should receive a test result.
3. The feedback should contain `expected: <printable> but was: <non-print�able>`.


### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage
unchanged
